src/commands/open.ts

```diff
@@ -0,0 +1,101 @@
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as open from "open";
+
+import * as FirebaseError from "../error";
+import * as api from "../api";
+import * as Command from "../command";
+import * as logger from "../logger";
+import { promptOnce } from "../prompt";
+import * as requirePermissions from "../requirePermissions";
+import * as requireInstance from "../requireInstance";
+import * as utils from "../utils";
+
+interface Link {
+  name: string;
+  arg: string;
+  consoleUrl?: string;
+  url?: string;
+}
+
+const LINKS: Link[] = [
+  { name: "Analytics", arg: "analytics", consoleUrl: "/analytics" },
+  { name: "Authentication: Providers", arg: "auth", consoleUrl: "/authentication/providers" },
+  { name: "Authentication: Users", arg: "auth:users", consoleUrl: "/authentication/users" },
+  { name: "Crash Reporting", arg: "crash", consoleUrl: "/monitoring" },
+  { name: "Database: Data", arg: "database", consoleUrl: "/database/data" },
+  { name: "Database: Rules", arg: "database:rules", consoleUrl: "/database/rules" },
+  { name: "Docs", arg: "docs", url: "https://firebase.google.com/docs" },
+  { name: "Dynamic Links", arg: "links", consoleUrl: "/durablelinks" },
+  { name: "Hosting: Deployed Site", arg: "hosting:site" },
+  { name: "Hosting", arg: "hosting", consoleUrl: "/hosting/main" },
+  { name: "Notifications", arg: "notifications", consoleUrl: "/notification" },
+  { name: "Project Dashboard", arg: "dashboard", consoleUrl: "/overview" },
+  { name: "Project Settings", arg: "settings", consoleUrl: "/settings/general" },
+  { name: "Remote Config: Conditions", arg: "config:conditions", consoleUrl: "/config/conditions" },
+  { name: "Remote Config", arg: "config", consoleUrl: "/config" },
+  { name: "Storage: Files", arg: "storage", consoleUrl: "/storage/files" },
+  { name: "Storage: Rules", arg: "storage:rules", consoleUrl: "/storage/rules" },
+  { name: "Test Lab", arg: "testlab", consoleUrl: "/testlab/histories/" },
+];
+
+const CHOICES = _.map(LINKS, "name");
+
+export default new Command("open [link]")
+  .description("quickly open a browser to relevant project resources")
+  .before(requirePermissions)
+  .before(requireInstance)
+  .action(
+    async (linkName: string, options: any): Promise<void> => {
+      let link = _.find(LINKS, { arg: linkName });
+      if (linkName && !link) {
+        throw new FirebaseError(
+          "Unrecognized link name. Valid links are:\n\n" + _.map(LINKS, "arg").join("\n")
+        );
+      }
+
+      if (!link) {
+        const name = await promptOnce({
+          type: "list",
+          message: "What link would you like to open?",
+          choices: CHOICES,
+        });
+        link = _.find(LINKS, { name });
+      }
+      if (!link) {
+        throw new FirebaseError(
+          "Unrecognized link name. Valid links are:\n\n" + _.map(LINKS, "arg").join("\n")
+        );
+      }
+
+      let url;
+      if (link.consoleUrl) {
```

Comment:
I feel like this whole if/else block can be simplified. For example, docs with url `if (link.url)` doesnt have to be a special case, and neither does `link.arg === "functions"`. As far as I can tell all of these are just taking some domain and adding specific paths to it to make the correct URL. Can we have a general method to do this with different base urls instead?